### PR TITLE
Adopt otelcol 0.70.0

### PR DESCRIPTION
**Pull request by rmfitzpatrick**
Updates deps and adjusts deprecated method.

### CODE REVIEW

**Review by atoulme — APPROVED** [review]
Changes look good. Approved, pending CI passes.

File: tests/receivers/oracledb/testdata/resource_metrics/all.yaml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 resource_metrics:
   - attributes:
-      oracledb.instance.name: localhost:1521
+      oracledb.instance.name: localhost:1521/XE
```
Comment (rmfitzpatrick) [inline tests/receivers/oracledb/testdata/resource_metrics/all.yaml:3]: per https://github.com/open-telemetry/opentelemetry-collector-contrib/commit/06d5dc53634a7859f84ed059f91af1ae62fce2d9

**Review by dmitryax — APPROVED** [review]
LGTM

File: CHANGELOG.md [hunk 2]
```diff
@@ -2,6 +2,19 @@
 
 ## Unreleased
 
+## v0.70.0
+
+This Splunk OpenTelemetry Collector release includes changes from the [opentelemetry-collector v0.70.0](https://github.com/open-telemetry/opentelemetry-collector/releases/tag/v0.70.0) and the [opentelemetry-collector-contrib v0.70.0](https://github.com/open-telemetry/opentelemetry-collector-contrib/releases/tag/v0.70.0) releases where appropriate.
+
+### 💡 Enhancements 💡
+
+- Initial [Discovery properties provider](https://github.com/signalfx/splunk-otel-collector/pull/2494) and config incorporation.
```
Comment (aunshc) [inline CHANGELOG.md:11]:
```suggestion
- Initial [Discovery properties provider](https://github.com/signalfx/splunk-otel-collector/pull/2494) and config incorporation for the discovery mode.
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -2,6 +2,19 @@
...
+- Third-party dependency updates.
+
+### 🧰 Bug fixes 🧰
+
+- [Address SignalFx exporter deferred metadata client initialization](https://github.com/open-telemetry/opentelemetry-collector-contrib/commit/f607cb47c8d972febb9d9d215e0029b3e8cb9884) causing [issues in the Smart Agent receiver](https://github.com/signalfx/splunk-otel-collector/issues/2508).
```
Comment (aunshc) [inline CHANGELOG.md:16]:
```suggestion
- [Addressed SignalFx exporter deferred metadata client initialization](https://github.com/open-telemetry/opentelemetry-collector-contrib/commit/f607cb47c8d972febb9d9d215e0029b3e8cb9884) causing [issues in the Smart Agent receiver](https://github.com/signalfx/splunk-otel-collector/issues/2508).
```

**Pull request merged**